can/viewer.py

```diff
@@ -300,6 +300,18 @@ def draw_can_bus_message(self, msg, sorting=False):
                     else:
                         values_list.append(str(x))
                 values_string = " ".join(values_list)
+
+                try:
+                    previousLength = self.ids[key]["previous_values_string_length"]
+                except KeyError:
+                    previousLength = 0
+
+                # Save length of string before adding spaces, such that the line is only blanked when necessary
+                self.ids[key]["previous_values_string_length"] = len(values_string)
+
+                if len(values_string) < previousLength:
+                    values_string += " " * (previousLength - len(values_string))
```

Comment:
```suggestion
                    values_string += " " * (self.x- len(values_string))
```

Wouldn't it be easier to fill all available space? Then you don't have to keep track of line length.
I didn't test this.

Reply:
I have included this suggestion, but have kept saving the length of the values_string to enable testing this behaviour.